hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1747,7 +1744,7 @@ public int getAsyncClusterMaxCommits() {
   }
 
   public String getPayloadClass() {
-    return getString(HoodiePayloadConfig.PAYLOAD_CLASS_NAME);
+    return RecordPayloadType.getPayloadClassName(this);
```

Comment:
Have you checked if all existing logic of getting the payload class go through this method?  Since this logic now returns the default payload class from the method instead of the config default value, we need to make sure any logic using `getString(HoodiePayloadConfig.PAYLOAD_CLASS_NAME)` should migrate to this API too, to avoid possible null or empty payload class name.

Reply:
I looked at the usage of `HoodiePayloadConfig.PAYLOAD_CLASS_NAME` and the logic looks to be working.